libcnb-package/src/lib.rs

```diff
@@ -259,9 +259,11 @@ pub fn get_buildpack_target_dir(
     buildpack_id: &BuildpackId,
     target_dir: &Path,
     is_release: bool,
+    target_triple: &str,
 ) -> PathBuf {
     target_dir
         .join("buildpack")
```

Comment:
I was wondering if we should revisit the `target/buildpack/...` prefix, given that:
(a) we're making a breaking change to the path already
(b) the GitHub Actions Rust cache action caches the whole `target/` directory (minus a few things it cleans up), when we don't need/really want the `target/buildpack/...` directory being cached
(c) it's more time consuming/error-prone to tab-complete a path under `target/...` (vs an otherwise empty directory) when wanting to run a buildpack locally and hand-typing `pack build --builpack target/buildpack/...` - since `target/` contains many other files/directories
(d) the path is now getting quite deep

eg: Using something like:
`dist/x86_64-unknown-linux-musl/debug/some-org_with-buildpack`

...instead of:
`target/buildpack/x86_64-unknown-linux-musl/debug/some-org_with-buildpack`

If not, then that's ok too. We could fix (b) via adding manual `target/` cleanup steps to our CI configs, and could improve the UX for (c) by implementing #484 instead.

Reply:
This makes sense to me. I think `target/` was only picked to just have a single output directory. I think not mixing it with the rust output directory makes sense, but it does mean that `dist` becomes part of the interface which I think is fine.

Reply:
Using a separate directory would also mean adding configuration options for said directory since it no longer can be configured by the standard cargo config options. Implicitly writing `dist` to the same (parent) directory `target` is in doesn't work reliably since that directory might be, for example, read-only.

@edmorley would you be fine approving this PR in the current state and I'll add another PR that moves the output to a separate directory that defaults to `$WORKSPACE_ROOT/dist/`? This way, we keep the changes grouped together and if we merge both before release there would be only one breaking change event for users.

Reply:
Yeah that sounds great, thank you :-)

Reply:
Also, feel free to switch to something other than `dist/` if you can think of a default better name.